enterprise/backend/src/metabase_enterprise/data_editing/api.clj

```diff
@@ -313,6 +314,35 @@
   (let [include-cols [:token :table_id :creator_id]]
     {:tokens (t2/select (into [:table_webhook_token] include-cols) :table_id table-id)}))
 
+(api.macros/defendpoint :get "/action"
+  "Lists all available actions and their parameters for a context.
+  Only supported context is from a dashboard editable right now, the dashcard-id must be supplied
+  as a query parameter.
+  Only returns actions the user has access to.
+  Return type is the same as GET /api/action"
+  [{:as _route-params}
+   {:keys [dashcard-id]} :- [:map
+                             [:dashcard-id ms/PositiveInt]]]
+  (let [dashcard        (api/check-404 (t2/select-one :model/DashboardCard :id dashcard-id))
+        table-id        (:table_id (:visualization_settings dashcard))
+        all-actions     (actions/select-actions nil :type "query")
+        model-ids       (sort (distinct (keep :model_id all-actions)))
+        coll-filter     (collection/visible-collection-filter-clause)
+        models          (when (seq model-ids)
+                          (t2/select :model/Card
+                                     {:where [:and
+                                              [:in :id model-ids]
+                                              [:= :type "model"]
+                                              [:= :archived false]
+                                              coll-filter]}))
+        table-model-ids (->> models
+                             (filter #(= table-id (:table_id %)))
+                             (map :id)
+                             set)
+        actions'        (->> all-actions
+                             (filter #(contains? table-model-ids (:model_id %))))]
```

Comment:
Pet hate is using threading macros with a single threaded form :-)

Reply:
good spot

Reply:
We could add a kondo for this maybe